src/main/java/org/nuxeo/wopi/rest/FilesResource.java

```diff
@@ -273,6 +276,66 @@ protected Object unlockOrRefresh(String fileId, String lock, boolean unlock) {
         }
     }
 
+    /**
+     * Implements the RenameFile operation.
+     * <p>
+     * See <a href="https://wopi.readthedocs.io/projects/wopirest/en/latest/files/RenameFile.html"></a>.
+     */
+    public Object renameFile(String fileId, String lock, String requestedName) {
+        NuxeoPrincipal principal = (NuxeoPrincipal) request.getUserPrincipal();
+        try (CloseableCoreSession session = createCoreSession(principal)) {
+            DocumentRef ref = new IdRef(fileId);
+            if (!session.exists(ref)) {
+                return Response.status(Response.Status.NOT_FOUND).build();
+            }
+
+            DocumentModel doc = session.getDocument(ref);
+            Blob blob = getMainBlob(doc);
+            if (blob == null) {
+                return Response.status(Response.Status.NOT_FOUND).build();
+            }
+
+            String nameJSON = "{ \"Name\": \"" + requestedName + "\" }";
+            if (!doc.isLocked()) {
+                if (!session.hasPermission(doc.getRef(), SecurityConstants.WRITE_PROPERTIES)) {
+                    // cannot rename blob
+                    return Response.status(Response.Status.CONFLICT).build();
```

Comment:
Is it expected to return a conflict instead a forbidden ?

Reply:
WOPI does not expect forbidden.